### PR TITLE
[IE CLDNN] Removed unused primitives and related structures

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
Things removed by this patch:
- Primitives required for training and related kernels
- Remaining int8 kernel written for old pipeline
- Related structures that are not required anymore

Size reduction 12863800 -> 11393208

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/deconvolution.cpp [hunk 1]
```diff
@@ -240,22 +236,12 @@ deconvolution_inst::typed_primitive_inst(network_impl& network, deconvolution_no
...
-                                  "input feature maps number",
-                                  filter_inst.size.feature[0],
-                                  "Weights/ifm mimsmatch");
-        }
+        CLDNN_ERROR_LESS_THAN(node.id(),
+                                "Weights feature maps number",
+                                (input_inst.size.feature[0] - input_offset.feature[0]) / split,
+                                "input feature maps number",
+                                filter_inst.size.feature[0],
+                                "Weights/ifm mimsmatch");
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/src/deconvolution.cpp:244]: Alignment

**Comment by vladimir-paramuzov** [comment]
> I suppose there is also unused function `void set_learning_params(const program_node& node, kernel_selector::training_params& params, bool use_momentum);` in kernel_selector_helper.h that can also be deleted

It's already removed in this PR

File: inference-engine/thirdparty/clDNN/src/deconvolution.cpp (hunk 1 above)
Comment (vladimir-paramuzov) [reply inference-engine/thirdparty/clDNN/src/deconvolution.cpp]: Fixed

**Comment by sshlyapn** [comment]
> > I suppose there is also unused function `void set_learning_params(const program_node& node, kernel_selector::training_params& params, bool use_momentum);` in kernel_selector_helper.h that can also be deleted
> 
> It's already removed in this PR

Oh, sorry, I didn't catch it with Ctrl+F

**Pull request merged**